generators/add-action/customer-profile/templates/getExperienceEvents.test.js

```diff
@@ -0,0 +1,87 @@
+<% } %>
+* <license header>
+*/
+
+jest.mock('@adobe/aio-sdk', () => ({
+  Core: {
+    Logger: jest.fn()
+  }
+}))
+
+jest.mock('@adobe/aio-lib-customer-profile', () => ({
+  CustomerProfileSDK: {
+    init: jest.fn()
+  }
+}))
+
+const { Core } = require('@adobe/aio-sdk')
+const CustomerProfileSDK = require('@adobe/aio-lib-customer-profile')
+const mockCustomerProfileInstance = { getExperienceEvents: jest.fn() }
+const mockLoggerInstance = { info: jest.fn(), debug: jest.fn(), error: jest.fn() }
+Core.Logger.mockReturnValue(mockLoggerInstance)
+CustomerProfileSDK.init.mockResolvedValue(mockCustomerProfileInstance)
+
+const action = require('./<%= actionRelPath %>')
+
+beforeEach(() => {
+  CustomerProfileSDK.init.mockClear() // only clears calls stats
+  mockCustomerProfileInstance.getExperienceEvents.mockReset() // clears calls + mock implementation
+
+  Core.Logger.mockClear()
+  mockLoggerInstance.info.mockReset()
+  mockLoggerInstance.debug.mockReset()
+  mockLoggerInstance.error.mockReset()
+})
+const fakeRequestParams = { tenantId: 'fakeId', iMSOrgId: 'fakeIMSOrgId', apiKey: 'fakeKey', __ow_headers: { authorization: 'Bearer fakeToken' } }
+describe('<%= actionName %>', () => {
+  test('main should be defined', () => {
+    expect(action.main).toBeInstanceOf(Function)
+  })
+  test('should set logger to use LOG_LEVEL param', async () => {
+    await action.main({ ...fakeRequestParams, LOG_LEVEL: 'fakeLevel' })
+    expect(Core.Logger).toHaveBeenCalledWith(expect.any(String), { level: 'fakeLevel' })
+  })
+  test('CustomerProfileSDK should be initialized with input credentials', async () => {
+    await action.main({ ...fakeRequestParams, otherParam: 'fake4' })
+    expect(CustomerProfileSDK.init).toHaveBeenCalledWith('fakeId', 'fakeKey', 'fakeIMSOrgId', 'fakeToken')
+  })
+  test('should return an http response with CustomerProfile API profile', async () => {
+    const fakeResponse = { profiles: 'fake' }
+    mockCustomerProfileInstance.getExperienceEvents.mockResolvedValue(fakeResponse)
+    const response = await action.main(fakeRequestParams)
+    expect(response).toEqual({
+      statusCode: 200,
+      body: fakeResponse
+    })
+  })
+  test('if there is an error should return a 500 and log the error', async () => {
+    const fakeError = new Error('fake')
+    mockCustomerProfileInstance.getExperienceEvents.mockRejectedValue(fakeError)
+    const response = await action.main(fakeRequestParams)
+    expect(response).toEqual({
+      error: {
+        statusCode: 500,
+        body: { error: 'server error' }
+      }
+    })
+    expect(mockLoggerInstance.error).toHaveBeenCalledWith(fakeError)
+  })
+  test('missing input request parameters, should return 400', async () => {
+    const response = await action.main({})
+    expect(response).toEqual({
+      error: {
+        statusCode: 400,
+        body: { error: 'missing header(s) \'authorization\' and missing parameter(s) \'tenantId,apiKey,iMSOrgId\'' }
```

Comment:
this test is failing in my generated app, the order of the  missing parameters in the string is different

Reply:
Updated.  I've replaced current example method with simpler one.